### PR TITLE
Cleaning up `moja_global_docs`

### PR DESCRIPTION
**Pull request by HarshCasper**
## Description

This PR:

- Formats all the Sphinx documentation 
- Adds setup instructions on README 
- Remove outdated information, GSoD references
- Add a Makefile for build setup and cleaning 
- Adds Sphinx GH Actions as part of the CI

### CODE REVIEW

**Review by aornugent — APPROVED** [review]
I haven't read every change in this PR because the new formatting has created very large diffs. You might like to get additional reviews from the Docs team?

Can you please check that the Bots section is relevant?

File: docs/DeveloperWorkflow/manual_testing_pr.rst [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-::
+\* Checkout to the Pull Request branch by running this command where
+<author> is the GitHub username of the PR author :
 
-git clone https://github.com/<author>/<repository-name>/
-git checkout <branch-name>
+git clone https://github.com/\ <author>/<repository-name>/ git checkout
+<branch-name>
 
-* Run the application locally and try to verify if the issue raised has been solved completely by the Pull Request changes.
```
Comment (aornugent) [inline docs/DeveloperWorkflow/manual_testing_pr.rst:14]: This section should be formatted as code

File: docs/DevelopmentSetup/git_and_github_guide.rst [hunk 2]
```diff
@@ -1,129 +1,92 @@
...
-    git add <file1> <file2>
-    git commit -amend
+This section will show you step-by-step how to make a contribution to
+FLINT using git.
 
-* Push this onto your feature branch but this time with force flag. This will update the pull request automatically. The reviewer won't be notified about this updation, so leave a comment in your pull request if you want a review.
-::
+-  FLINT stable branch is **develop**. Releases are scheduled
+   periodically when codebase is production-ready and develop is merged
+   into master. develop branch
```
Comment (aornugent) [inline docs/DevelopmentSetup/git_and_github_guide.rst:86]: This looks like an unfinished sentence

File: docs/DeveloperWorkflow/manual_testing_pr.rst (hunk 1 above)
Comment (HarshCasper) [reply docs/DeveloperWorkflow/manual_testing_pr.rst:14]: Fixed!

File: docs/DevelopmentSetup/git_and_github_guide.rst (hunk 2 above)
Comment (HarshCasper) [reply docs/DevelopmentSetup/git_and_github_guide.rst]: Done!

File: docs/DevelopmentSetup/visual_studio_win_example.rst [hunk 3]
```diff
@@ -74,219 +78,267 @@ Commands to run cmake for the spatial simulations:
...
-* Open the visual studio solution that CMake created at ``C:\Development\moja-global\FLINT\Source\build\moja.sln``
-
-* Build the debug configuration ``ALL_BUILD`` target by right clicking the ``ALL_BUILD`` node and selecting ``Build``.
+-  Open the visual studio solution that CMake created at
+   ``C:\Development\moja-global\FLINT\Source\build\moja.sln``
+-  Build the debug configuration ``ALL_BUILD`` target by right clicking
+   the ``ALL_BUILD`` node and selecting ``Build``.
 
 .. figure:: ../images/installation_vs2019_flint.example/VS2019_buildall.jpeg
-  :width: 600
```
Comment (shubhamkarande13) [inline docs/DevelopmentSetup/visual_studio_win_example.rst:88]: Screenshots not visible!
https://moja-global-documentation--73.org.readthedocs.build/en/73/DevelopmentSetup/visual_studio_win_example.html

File: docs/DevelopmentSetup/visual_studio_win_example.rst [hunk 4]
```diff
@@ -74,219 +78,267 @@ Commands to run cmake for the spatial simulations:
...
+   page
+   :align: center
+   :width: 600px
 
-  Navigating to Visual Studio Debugging ``ALL_BUILD`` properties page
+   Navigating to Visual Studio Debugging ``ALL_BUILD`` properties page
 
 .. figure:: ../images/installation_vs2019_flint.example/Step4.png
-  :width: 600
-  :align: center
```
Comment (shubhamkarande13) [inline docs/DevelopmentSetup/visual_studio_win_example.rst:96]: Screenshot not visible!
https://moja-global-documentation--73.org.readthedocs.build/en/73/DevelopmentSetup/visual_studio_win_example.html

File: docs/DevelopmentSetup/docker_installation_example.rst [hunk 5]
```diff
@@ -1,101 +1,89 @@
...
+   # For Linux
+   docker run --rm -v $(pwd)/Run_Env:/usr/local/run_env -ti moja/flint.example:bionic bash -c "cd /usr/local/run_env/; moja.cli --config config/point_rothc_example.json --config config/libs.base_rothc.simple.json --logging_config logging.debug_on.conf"
 
-    # For Windows
-    docker run --rm -v %cd%/Run_Env:/usr/local/run_env -ti moja/flint.example:bionic bash -c "cd /usr/local/run_env/; moja.cli --config config/point_rothc_example.json --config config/libs.base_rothc.simple.json --logging_config logging.debug_on.conf"
+   # For Windows
+   docker run --rm -v %cd%/Run_Env:/usr/local/run_env -ti moja/flint.example:bionic bash -c "cd /usr/local/run_env/; moja.cli --config config/point_rothc_example.json --config config/libs.base_rothc.simple.json --logging_config logging.debug_on.conf"
 
 .. figure:: ../images/installation_docker/step2_docker.png
-  :width: 600
```
Comment (shubhamkarande13) [inline docs/DevelopmentSetup/docker_installation_example.rst:53]: Screenshot not visible!
https://moja-global-documentation--73.org.readthedocs.build/en/73/DevelopmentSetup/windows_installation.html#running-the-project

**Review by shubhamkarande13 — CHANGES_REQUESTED** [review]
Please fix the missing screenshots! 😅

File: docs/GCBMDevelopmentSetup/prerequisites.rst [hunk 6]
```diff
@@ -1,85 +1,101 @@
...
 
 Install the C++ packages required to run GCBM and supporting tools:
 
-* Double-click ``tools\VC_redist\install_vcredist.bat``
+-  Double-click ``tools\VC_redist\install_vcredist.bat``
 
 .. figure:: ../images/installation_gcbm/image6.png
-  :width: 600
-  :align: center
-  :alt: Installing the C++ packages required to run GCBM and supporting tools
```
Comment (shubhamkarande13) [inline docs/GCBMDevelopmentSetup/prerequisites.rst:72]: Screenshots missing!
https://moja-global-documentation--73.org.readthedocs.build/en/73/GCBMDevelopmentSetup/prerequisites.html#visual-c-redistributable-packages

**Review by shubhamkarande13 — CHANGES_REQUESTED** [review]

File: docs/DeveloperWorkflow/automated_checks.rst [hunk 7]
```diff
@@ -1,111 +1,85 @@
...
+If the coverage percentage remains same or higher than before, the check
+passes. Else if the percentage becomes lower, the check fails.
 
-* In order to debug and understand the failure of this check you may click on the **Details** tab of the ``codecov/project`` check under Checks tab.
+-  In order to debug and understand the failure of this check you may
+   click on the **Details** tab of the ``codecov/project`` check under
+   Checks tab.
 
 .. figure:: ../images/all_checks.png
-  :width: 600
```
Comment (shubhamkarande13) [inline docs/DeveloperWorkflow/automated_checks.rst:54]: Is this section really not required now?

**Review by shubhamkarande13 — CHANGES_REQUESTED** [review]

File: docs/DeveloperWorkflow/bots_and_integrations.rst [hunk 8]
```diff
@@ -1,103 +1,84 @@
...
+~~~~~~~~~~~~
 
+If you wish to configure or modify the bot settings according to the
+needs of the repository, you may update the ``.all-contributorsrc`` JSON
+file. The data used to generate the contributors list will be stored in
+there, and you can configure how you want ``@all-contributors`` to
+generate the list. Here are more details on the `configuration options`_
+available.
 
 Botmojaglobal
```
Comment (chicken-biryani) [inline docs/DeveloperWorkflow/bots_and_integrations.rst:56]: I don't think this bot is functioning currently. Putting details about something that's not currently functioning may be misleading.

**Review by chicken-biryani — COMMENTED** [review]
So:
1) The formatting looks good overall.
2) The statements in the README can be made a little more compact. You can use [easybib.com](https://www.easybib.com/grammar-and-plagiarism/) or any other tool. I think you've opted for a conversation style that looks good.

That's all from my side for now.

**Comment by shubhamkarande13** [comment]
Thank you Shloka! 
Harsh, I think we need some more time and efforts in ironing out all the dead links in this PR.
Amazing work with the cleanup!  🚀

File: docs/DevelopmentSetup/FLINT.example_installation.rst [hunk 9]
```diff
@@ -1,33 +1,46 @@
...
-* **Chapman richards** - Point sample
-* **Chapman richards** - Spatial sample
+-  **Test Module sample** - Point level
+-  **RothC sample** - Point level
+-  **Chapman richards** - Point sample
+-  **Chapman richards** - Spatial sample
 
-The FLINT.example repository is available `here <https://github.com/moja-global/FLINT.Example>`_ under the moja global organisation on GitHub. Before proceeding to the instuctions for installing FLINT.example, please follow the following steps to clone this repository on your fork:
+The FLINT.example repository is available `here`_ under the moja global
+organisation on GitHub. Before proceeding to the instuctions for
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DevelopmentSetup/FLINT.example_installation.rst:31]: Fixed!

File: docs/DeveloperWorkflow/reviewing.rst [hunk 10]
```diff
@@ -1,60 +1,104 @@
...
+added cover edge cases and test various scenarios instead of the most
+commonly used ones.
 
 Design
 ======
 
-If the code changes introduce design changes in terms of UI/UX, please ask for screnshots/gif supporting this change. Feel free to ask for screenshots on devices with different screen sizes such as mobile/tablet view to get a better idea if the design promotes responsiveness.
-Responsive design helps the application to be more accessible to people thus reaching a wider audience.
+If the code changes introduce design changes in terms of UI/UX, please
+ask for screnshots/gif supporting this change. Feel free to ask for
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/reviewing.rst:89]: Fixed!

File: docs/DeveloperWorkflow/reviewing.rst [hunk 11]
```diff
@@ -1,60 +1,104 @@
...
-* Make sure the pull request is directed to the correct target branch.
-* Make sure the feature branch, the commit message and the pull request title is concise and appropriate. For providing the author instructions on how to do that, please direct them to our `commit message section of Code Contribution Best Practices <../contributing/code_contribution_best_practices.html#commit-message-guidelines>`_.
-* Make sure the commits added in the pull request are clean and few in number. This will help us in keeping the commit history clean.
-* If the number of files changed in a pull request is quite high, it might be a good idea to ask the author to split the pull request into smaller ones if possible.
-* Check whether all PR checks are passing or not. If not, you may also help the author debug these checks and help them contribute.
+-  While suggesting changes in the pull requests, the key is to direct
+   your suggestions at the code and not at the **author**. This promotes
+   a healthy discussion instead of making someone feel that their code
+   wasn't upto the mark.
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/reviewing.rst:25]: Fixed!

File: docs/DeveloperWorkflow/reviewing.rst [hunk 12]
```diff
@@ -1,60 +1,104 @@
...
-* Make sure the pull request is directed to the correct target branch.
-* Make sure the feature branch, the commit message and the pull request title is concise and appropriate. For providing the author instructions on how to do that, please direct them to our `commit message section of Code Contribution Best Practices <../contributing/code_contribution_best_practices.html#commit-message-guidelines>`_.
-* Make sure the commits added in the pull request are clean and few in number. This will help us in keeping the commit history clean.
-* If the number of files changed in a pull request is quite high, it might be a good idea to ask the author to split the pull request into smaller ones if possible.
-* Check whether all PR checks are passing or not. If not, you may also help the author debug these checks and help them contribute.
+-  While suggesting changes in the pull requests, the key is to direct
+   your suggestions at the code and not at the **author**. This promotes
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/reviewing.rst:23]: Done!

File: docs/DeveloperWorkflow/reviewing.rst [hunk 13]
```diff
@@ -1,60 +1,104 @@
...
-Here's a guide on how to get started!
+moja global welcomes all contributors to review each others pull
+requests and suggest changes. If you have been contributing to moja
+global repositories, we highly encourage you to review pull requests as
+well. Here's a guide on how to get started!
 
-Quoting GitHub documentation: “Reviews allow for discussion of proposed changes and help ensure that the changes meet the repository’s contributing guidelines and other quality standards.“.
+Quoting GitHub documentation: “Reviews allow for discussion of proposed
+changes and help ensure that the changes meet the repository’s
+contributing guidelines and other quality standards.“.
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/reviewing.rst:13]: Done!

File: docs/DeveloperWorkflow/repository_maintenance.rst [hunk 14]
```diff
@@ -1,248 +1,339 @@
...
+   :width: 600px
 
-As moja global is an open-source project and community, consider using the project boards to update other team members, and encourage participation inside of GitHub issues and pull requests.
+As moja global is an open-source project and community, consider using
+the project boards to update other team members, and encourage
+participation inside of GitHub issues and pull requests.
 
-In the meanwhile, we also consider using the project boards for development. It also helps remind us and other core contributors to spend 5 minutes each day updating progress as needed.
+In the meanwhile, we also consider using the project boards for
+development. It also helps remind us and other core contributors to
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/repository_maintenance.rst:338]: Done!

File: docs/DeveloperWorkflow/repository_maintenance.rst [hunk 15]
```diff
@@ -1,248 +1,339 @@
...
-  :alt: Create a new project board
-
+   :alt: Create a new project board
+   :align: center
+   :width: 600px
 
-* After creating the project board, you may make adjustments to it as needed. You can create new columns, set up automation and add pre-existing GitHub issues and pull requests to the project board.
+-  After creating the project board, you may make adjustments to it as
+   needed. You can create new columns, set up automation and add
+   pre-existing GitHub issues and pull requests to the project board.
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/repository_maintenance.rst:253]: Done!

File: docs/DeveloperWorkflow/repository_maintenance.rst [hunk 16]
```diff
@@ -1,248 +1,339 @@
...
+=================================================
 
-After you set up a project board, you need to populate it with issues and pull requests and keep updating the board on a regular basis.
+After you set up a project board, you need to populate it with issues
+and pull requests and keep updating the board on a regular basis.
 
-The ``Basic kanban`` template offers the following columns for every issue/pull request. You may classify your issues/pull requests into one of according to the criteria below.
+The ``Basic kanban`` template offers the following columns for every
+issue/pull request. You may classify your issues/pull requests into one
+of according to the criteria below.
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/repository_maintenance.rst:267]: Done!

File: docs/DeveloperWorkflow/repository_maintenance.rst [hunk 17]
```diff
@@ -1,248 +1,339 @@
...
+developers to use pull requests too.
+
+Follow contributing guidelines The contributing guidelines for all
+MagicBox projects live in the xxxx repository.
+
+Always follow these contributing guidelines when working in the project.
+These are the standards and rules we ask the community to follow when
+contributing. As project maintainers, it is our responsibility to hold
+ourselves to the same standards we ask of others. Thus, always make sure
+current development practices are in-line with what our guidelines.
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/repository_maintenance.rst:176]: Done!

File: docs/DeveloperWorkflow/repository_maintenance.rst [hunk 18]
```diff
@@ -1,248 +1,339 @@
...
+labels). Add labels from the metadata sub-menu when you open an issue or
+pull request. Making pull requests All major changes to the project
+should always be made through a pull request (PR). Pull requests are
+like a registry of changes for a project. It is easy for someone to see
+what is going in and out of a project. Outside contributors will always
+have to make pull requests, so it is good practice for core / trusted
+developers to use pull requests too.
+
+Follow contributing guidelines The contributing guidelines for all
+MagicBox projects live in the xxxx repository.
```
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/DeveloperWorkflow/repository_maintenance.rst:170]: Fixed! These parts exactly concern the old docs refactor that I didn't cover during this cleanup.

File: docs/DeveloperWorkflow/automated_checks.rst (hunk 7 above)
Comment (HarshCasper) [reply docs/DeveloperWorkflow/automated_checks.rst:54]: It was missed during formatting. Added it back.

**Review by shubhamkarande13 — APPROVED** [review]

**Comment by HarshCasper** [comment]
@all-contributors bot please add @HarshCasper for code docs review maintenance
